src/artifact/parser.cpp

```diff
@@ -48,6 +48,12 @@ namespace payload = mender::artifact::v3::payload;
 
 ExpectedArtifact VerifyEmptyPayloadArtifact(
 	Artifact &artifact, lexer::Lexer<token::Token, token::Type> &lexer) {
+	if (artifact.header.subHeaders.size() == 0) {
```

Comment:
Why not
```suggestion
	if (artifact.header.subHeaders.size() != 1) {
```

This special Artifact is under our control for the bootstrap functionality, so we can be stricter in what to expect.